jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -159,6 +173,13 @@ def _on_subset_update(self, msg):
         elif sbst.label == self.bg_subset_selected and sbst.data.label == self.dataset_selected:
             self._bg_subset_selected_changed()
 
+    def _on_link_update(self, msg):
+        if self.dataset_selected == '' or self.subset_selected == '':
+            return
+
+        # Force background auto-calculation to update when linking has changed.
+        self._bg_subset_selected_changed()

```

Comment:
Good point. I didn't test the annulus case. I'll do it when I get back, or feel free to push commits here while I am gone if this is in a rush.

If we end up needing `_subset_selected_changed()`, then I think we don't have to call `_bg_subset_selected_changed()` because the former also triggers the latter.

Reply:
I pushed a commit to also fix the annulus case. Please re-review. Good catch. Thanks!